### PR TITLE
Fix allocations

### PR DESCRIPTION
**Pull request by RblSb**

### CODE REVIEW

**Comment by RblSb** [comment]
Should be this `field = new Matrix` case be reported as inlining bug?

**Comment by RblSb** [comment]
Rip allocs. Closes #462 and also #1035.

File: Sources/kha/graphics2/Graphics.hx [hunk 1]
```diff
@@ -235,12 +228,14 @@ class Graphics {
 	#end
 
 	private var transformations: Array<FastMatrix3>;
+	private var transformationId: Int;
```
Comment (sh-dave) [inline Sources/kha/graphics2/Graphics.hx]: rename to `index` instead of `id`?

File: Sources/kha/graphics2/Graphics.hx [hunk 2]
```diff
@@ -235,12 +228,14 @@ class Graphics {
 	#end
 
 	private var transformations: Array<FastMatrix3>;
+	private var transformationId: Int;
 	private var opacities: Array<Float>;
 	private var myFontSize: Int;
 
 	public function new() {
 		transformations = new Array<FastMatrix3>();
```
Comment (sh-dave) [inline Sources/kha/graphics2/Graphics.hx]: array comprehension ftw! `transformations = [FastMatrix3.identity()];`

Comment (sh-dave) [inline Sources/kha/graphics2/Graphics.hx]: array comprehension ftw! `transformations = [FastMatrix3.identity()];`

File: Sources/kha/graphics2/Graphics.hx [hunk 3]
```diff
@@ -235,12 +228,14 @@ class Graphics {
...
 
 	private var transformations: Array<FastMatrix3>;
+	private var transformationId: Int;
 	private var opacities: Array<Float>;
 	private var myFontSize: Int;
 
 	public function new() {
 		transformations = new Array<FastMatrix3>();
 		transformations.push(FastMatrix3.identity());
+		transformationId = transformations.length - 1;
```
Comment (sh-dave) [inline Sources/kha/graphics2/Graphics.hx]: isn't that like always 0, and would even be wrong if you actually pushed more than 1 default matrix? :)

**Comment by RblSb** [comment]
Yeah, I decided not to change some lines for a simpler diff, but let's dave deeper

**Comment by Joncom** [comment]
@RblSb  Why did you say this closes #1035?

https://github.com/Kode/Kha/pull/1044#issuecomment-490812343

Or was that before you simplified this pull request?

**Comment by RblSb** [comment]
You can close it already, I just mention this issue for Robert.

**Pull request merged**